### PR TITLE
impl(pubsub): add pull ack handler factory

### PR DESCRIPTION
**Pull request by alevenberg**
Part of https://github.com/googleapis/google-cloud-cpp/issues/13265

4/5 in a chain of PRs to add tracing for the settle span for the unary pull

1. https://github.com/googleapis/google-cloud-cpp/pull/13350
2. https://github.com/googleapis/google-cloud-cpp/pull/13353
3. https://github.com/googleapis/google-cloud-cpp/pull/13354
4. https://github.com/googleapis/google-cloud-cpp/pull/13358
5. https://github.com/googleapis/google-cloud-cpp/pull/13360

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13358)

### CODE REVIEW

File: google/cloud/pubsub/internal/pull_ack_handler_factory.cc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+pubsub::PullAckHandler CreatePullAckHandler(
+    CompletionQueue cq, std::weak_ptr<SubscriberStub> stub,
+    pubsub::Subscription const& subscription, std::string const& ack_id,
```
Comment (coryan) [inline google/cloud/pubsub/internal/pull_ack_handler_factory.cc:28]: `std::move()` for a `T const&` does not really help.  This function should probably be:

```suggestion
    pubsub::Subscription subscription, std::string  ack_id,
```

File: google/cloud/pubsub/internal/pull_ack_handler_factory.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+pubsub::PullAckHandler CreatePullAckHandler(
+    CompletionQueue cq, std::weak_ptr<SubscriberStub> stub,
+    pubsub::Subscription const& subscription, std::string const& ack_id,
```
Comment (coryan) [inline google/cloud/pubsub/internal/pull_ack_handler_factory.h:35]: Same change as in the `.cc` file.

**Review by coryan — APPROVED** [review]

File: google/cloud/pubsub/internal/pull_ack_handler_factory.h (hunk 2 above)
Comment (alevenberg) [inline google/cloud/pubsub/internal/pull_ack_handler_factory.h:35]:
```suggestion
    pubsub::Subscription subscription, std::string  ack_id,
```

File: google/cloud/pubsub/internal/pull_ack_handler_factory.cc [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+pubsub::PullAckHandler CreatePullAckHandler(
+    CompletionQueue cq, std::weak_ptr<SubscriberStub> stub,
+    pubsub::Subscription subscription, std::string  ack_id,
+    std::int32_t delivery_attempt, Options const& options) {
+  std::unique_ptr<pubsub::PullAckHandler::Impl> impl =
+      std::make_unique<pubsub_internal::DefaultPullAckHandler>(
+          std::move(cq), std::move(stub), options, std::move(subscription),
+          ack_id, delivery_attempt);
```
Comment (coryan) [inline google/cloud/pubsub/internal/pull_ack_handler_factory.cc:33]:
```suggestion
          std::move(ack_id), delivery_attempt);
```

**Review by coryan — APPROVED** [review]

**Pull request merged**